fedimint-api/src/config.rs

```diff
@@ -64,32 +63,36 @@ impl ClientConfig {
 ///
 /// Candidate for re-designing when the modularization effort is
 /// complete.
-pub struct ModuleConfigGenParams {
-    pub mint_amounts: Vec<Amount>,
-    pub bitcoin_rpc: BitcoindRpcCfg,
-
-    /// extra options for extra settings and modules
-    pub other: BTreeMap<String, serde_json::Value>,
-}
-
-impl ModuleConfigGenParams {
-    /// Default & fake config gen params for things like tests
-    ///
-    /// TODO: Possibly this does not belong here.
-    pub fn fake_config_gen_params() -> ModuleConfigGenParams {
-        ModuleConfigGenParams {
-            mint_amounts: [1, 10, 100, 1000, 10000, 100000, 1000000]
-                .into_iter()
-                .map(Amount::from_msats)
-                .collect(),
-            bitcoin_rpc: fedimint_api::config::BitcoindRpcCfg {
-                btc_rpc_address: "localhost".into(),
-                btc_rpc_user: "bitcoin".into(),
-                btc_rpc_pass: "bitcoin".into(),
-            },
-            other: Default::default(),
-        }
+#[derive(Debug, Clone, Default)]
+pub struct ConfigGenParams(BTreeMap<String, serde_json::Value>);
+
+impl ConfigGenParams {
+    pub fn new() -> ConfigGenParams {
+        ConfigGenParams::default()
+    }
+
+    /// Add params for a module
+    pub fn attach<P: ModuleConfigGenParams>(mut self, module_params: P) -> Self {
```

Comment:
Nice.

Reply:
I think I'll try to figure out better naming. I guess the `ServerConfigParams` will have a function `with_module<P>(params: P)` that will delegate to this function, so it should have the same name I guess.